manabi/lock.py

```diff
@@ -94,3 +117,125 @@ def get_lock_list(self, path, *, include_root, include_children, token_only):
                 include_children=include_children,
                 token_only=token_only,
             )
+
+
+class PostgresDict(MutableMapping):
+    def __init__(self, connection: psycopg2.extensions.connection):
+        self._connection = connection
+
+    @staticmethod
+    def decode_lock(lock):
+        # Encoding is not needed since a string is also valid for wsgidav
+        if "owner" in lock:
+            owner = lock["owner"]
+            if isinstance(owner, bytes):
+                lock["owner"] = lock["owner"].decode("UTF-8")
+
+    def cursor(self) -> psycopg2.extensions.cursor:
+        return self._connection.cursor()
+
+    def cleanup(self):
+        cursor = self.cursor()
+        cursor.execute("DELETE FROM manabi_lock;")
+
+    def __len__(self):
+        cursor = self.cursor()
+        cursor.execute("SELECT count(*) FROM manabi_lock")
+        return int(cursor.fetchone())
+
+    def __iter__(self):
+        cursor = self.cursor()
+        cursor.execute("SELECT token FROM manabi_lock")
+        for token in cursor.fetchall():
+            yield token[0]
+
+    def __delitem__(self, token):
+        cursor = self.cursor()
+        cursor.execute("DELETE FROM manabi_lock WHERE token = %s", (str(token),))
+
+    def __contains__(self, token):
+        cursor = self.cursor()
+        cursor.execute("SELECT * FROM manabi_lock WHERE token = %s", (str(token),))
```

Comment:
```suggestion
        cursor.execute("SELECT 1 FROM manabi_lock WHERE token = %s", (str(token),))
```
Avoids page read on disk, as it only hits the index this way

Reply:
Thanks, is fixed.